hibernate-core/src/test/java/org/hibernate/orm/test/jpa/criteria/basic/AggregationResultTest.java

```diff
@@ -87,6 +88,19 @@ public void testSumOfIntegers() {
 		em.close();
 	}
 
+	@Test
+	public void testTuplesWithSumOfIntegers() {
+		EntityManager em = getOrCreateEntityManager();
+		em.getTransaction().begin();
+		CriteriaQuery<Tuple> criteria = builder.createTupleQuery();
+		Root<Product> productRoot = criteria.from( Product.class );
+		criteria.select(builder.tuple( builder.sum( productRoot.get( Product_.quantity ) ) ));
+		Tuple sumResult = em.createQuery( criteria ).getSingleResult();
+		assertReturnType( Long.class, sumResult );
```

Comment:
Assuming that `Product.quantity` is of type `Integer`, the type of `builder.sum( productRoot.get( Product_.quantity ) )` is `Expression<Integer>`, and the `Tuple` should contain an `Integer` and not a `Long`.

So I believe this assertion is correct, and that there's no bug here.